src/main/java/rx/internal/operators/OperatorDistinctUntilChanged.java

```diff
@@ -62,7 +62,7 @@ public OperatorDistinctUntilChanged(Func2<? super U, ? super U, Boolean> compara
     
     @Override
     public Boolean call(U t1, U t2) {
-        return (t1 == t2 || (t1 != null && t1.equals(t2)));
+        return (t1 == t2 || (t2 != null && t2.equals(t1)));
```

Comment:
An example:
Object `t1` and `t2` are of the same type and contains an object `x1`.
Object `x1` is `null` in `t1` and is not null in `t2`
Object `t1` and `t2` override `equals` which in turn calls the `equals` of `x1`.
A `NullpointerException` is thrown from `t1` but not from `t2`.

This is a basic example but what I want to say is that applying `equals` on `t1` does not means that it will result the same the other way around.
That's the type of behavior changes which should be fixed on the user side of the rxjava lib but which can create confusion and can be avoided.